runtime/src/bank.rs

```diff
@@ -514,6 +548,136 @@ impl Bank {
+        genesis_config: &GenesisConfig,
+        fields: BankFields,
+    ) -> Self {
+        fn new<T: Default>() -> T {
+            T::default()
+        }
+        let mut bank = Self {
+            rc: bank_rc,
+            src: new(),
+            blockhash_queue: RwLock::new(fields.blockhash_queue),
+            ancestors: fields.ancestors,
+            hash: RwLock::new(fields.hash),
+            parent_hash: fields.parent_hash,
+            parent_slot: fields.parent_slot,
+            hard_forks: Arc::new(RwLock::new(fields.hard_forks)),
+            transaction_count: AtomicU64::new(fields.transaction_count),
+            tick_height: AtomicU64::new(fields.tick_height),
+            signature_count: AtomicU64::new(fields.signature_count),
+            capitalization: AtomicU64::new(fields.capitalization),
+            max_tick_height: fields.max_tick_height,
+            hashes_per_tick: fields.hashes_per_tick,
+            ticks_per_slot: fields.ticks_per_slot,
+            ns_per_slot: fields.ns_per_slot,
+            genesis_creation_time: fields.genesis_creation_time,
+            slots_per_year: fields.slots_per_year,
+            unused: genesis_config.unused,
+            slot: fields.slot,
+            epoch: fields.epoch,
+            block_height: fields.block_height,
+            collector_id: fields.collector_id,
+            collector_fees: AtomicU64::new(fields.collector_fees),
+            fee_calculator: fields.fee_calculator,
+            fee_rate_governor: fields.fee_rate_governor,
+            collected_rent: AtomicU64::new(fields.collected_rent),
+            rent_collector: fields.rent_collector,
+            epoch_schedule: fields.epoch_schedule,
+            inflation: Arc::new(RwLock::new(fields.inflation)),
+            stakes: RwLock::new(fields.stakes),
+            epoch_stakes: fields.epoch_stakes,
+            is_delta: AtomicBool::new(fields.is_delta),
+            message_processor: new(),
+            entered_epoch_callback: new(),
+            last_vote_sync: new(),
+            rewards: new(),
+            skip_drop: new(),
+            operating_mode: Some(genesis_config.operating_mode),
+            lazy_rent_collection: new(),
+        };
+        bank.finish_init();
+
+        // Sanity assertions between bank snapshot and genesis config
+        // Consider removing from serializable bank state ([Ref]BankFields) and initializing
+        // from the passed in genesis_config instead (as new()/new_with_paths() already do)
+        assert_eq!(
+            bank.hashes_per_tick,
+            genesis_config.poh_config.hashes_per_tick
+        );
+        assert_eq!(bank.ticks_per_slot, genesis_config.ticks_per_slot);
+        assert_eq!(
+            bank.ns_per_slot,
+            genesis_config.poh_config.target_tick_duration.as_nanos()
+                * genesis_config.ticks_per_slot as u128
+        );
+        assert_eq!(bank.genesis_creation_time, genesis_config.creation_time);
```

Comment:
@svenski123 (Thanks for continued work on this!) As a quick reminder, it seems that CI failed at this assertion:

https://buildkite.com/solana-labs/solana/builds/27132#3b380563-b5ce-418e-bb89-7a9012d83ae9/79-2158
```
failures:
    tests::V1_1_0::test_bank_forks_snapshot_n
    tests::V1_1_0::test_bank_forks_status_cache_snapshot_n
    tests::V1_2_0::test_bank_forks_snapshot_n
    tests::V1_2_0::test_bank_forks_status_cache_snapshot_n
```

Reply:
Yes it did fail, due to a bug in the bank_forks tests now fixed.